File: base-class-module/src/main/java/jingya/com/base_class_module/BaseUtils/ToastUtils.java
package jingya.com.base_class_module.BaseUtils;

import android.content.Context;
import android.view.Gravity;
import android.widget.Toast;

/**
 * @author Kuky
 */
public class ToastUtils {
    private static Toast mToast;

    public static void showToast(Context context, String msg) {
        showToast(context, msg, Toast.LENGTH_SHORT);
    }

    public static void showToast(Context context, String msg, int duration) {
        if (mToast == null)
            mToast = Toast.makeText(context, msg, duration);
        else {
            mToast.setText(msg);
            mToast.setDuration(duration);
        }
        mToast.show();
    }

    public static void showToastCenter(Context context, String msg) {
        showToastCenter(context, msg, Toast.LENGTH_SHORT);
    }

    public static void showToastCenter(Context context, String msg, int duration) {
        if (mToast == null)
            mToast = Toast.makeText(context, msg, duration);
        else {
            mToast.setText(msg);
            mToast.setDuration(duration);
        }
        mToast.setGravity(Gravity.CENTER, 0, 0);
        mToast.show();
    }

    public static void cancelToast() {
        if (mToast != null) {
            mToast.cancel();
            mToast = null;
        }
    }
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/BaseUtils/ToastUtils.kt
package com.kuky.baselib.baseUtils

import android.annotation.SuppressLint
import android.content.Context
import android.view.Gravity
import android.widget.Toast

/**
 * @author kuky
 * @description
 */
object ToastUtils {
    private var mToast: Toast? = null

    @SuppressLint("ShowToast")
    fun showToast(context: Context, msg: String, length: Int = Toast.LENGTH_SHORT) {
        if (mToast == null) {
            mToast = Toast.makeText(context, msg, length)
        } else {
            mToast!!.setText(msg)
        }

        mToast!!.show()
    }

    @SuppressLint("ShowToast")
    fun showCenterToast(context: Context, msg: String, length: Int = Toast.LENGTH_SHORT) {
        if (mToast == null) {
            mToast = Toast.makeText(context, msg, length)
        } else {
            mToast!!.setText(msg)
        }

        mToast!!.setGravity(Gravity.CENTER, 0, 0)
        mToast!!.show()
    }

    fun cancelToast() {
        if (mToast != null) {
            mToast!!.cancel()
            mToast = null
        }
    }
}